Xamarin.Forms.Core/Application.cs

```diff
@@ -76,6 +76,13 @@ public Page MainPage
 				if (_mainPage != null)
 				{
 					InternalChildren.Remove(_mainPage);
+					_mainPage.NavigationProxy.Inner = null;
+
+					// Set platform inner proxy to null
+					// otherwise the incoming page will make decisions based on the 
+					// previous application state
+					NavigationProxy.Inner = null;
```

Comment:
This is the biggest change here I'm wondering about the approach on

The problem I noticed is that this Inner is only set once the page is realized on the platform. 

This causes an issue for the incoming page because when the incoming page checks *Navigation.ModalStack* that call gets proxied to the previous platform *INavigation* instance. So it makes decisions based on modals/pages that aren't going to exist anymore